hypothesis-python/src/hypothesis/internal/reflection.py

```diff
@@ -303,6 +303,7 @@ def visit_Lambda(self, node):
 SPACE_PRECEDES_CLOSE_BRACKET = re.compile(r" \)")
 
 
+@lru_cache(maxsize=1024)
 def extract_lambda_source(f):
```

Comment:
I think ideally we would cache on `get_pretty_function_description`, but that is called with non-hashable types like classes. Potentially worth handling those cases early in `get_pretty_function_description` and caching the entire remaining logic for hashable/function types. That said, the only other logic is `inspect.isclass`, so it should be ok...

Reply:
either way seems fine to me 👍

Reply:
went with caching at the higher level because it set my mind slightly more at ease.

Reply:
Nevermind: 

```
FAILED hypothesis-python/tests/cover/test_filter_rewriting.py::test_isidentifier_filter_properly_rewritten[None] - TypeError: cannot create weak reference to 'method_descriptor' object
```

More trouble than it's worth, let's stick with caching at `extract_lambda_source`.